### ISSUE TITLE
Question about oneOf

### ISSUE THREAD

**Issue by SebastianStehle** [RicoSuter/NSwag#3260]
Hi,

I have the following schema: https://app.notifo.io/api/docs/index.html?url=/api/openapi.json

It works very great with NSwag and typescript/C# code generation, but many other code generators seem to have issues with oneOf+$ref and generate classes or empty interfaces for all nested objects.

Is this a problem of the code generators or something that can be improved?

**Comment by jeremyVignelles** [comment]
What's your question if it works well with NSwag?
oneOf is not something that is very easy to work with in a strong-typed language like C#.

**Comment by SebastianStehle** [comment]
I need a ruby and java client and these code generators seem to have issues.

**Comment by jeremyVignelles** [comment]
It might be because NSwag always generates a oneOf level, which is not expected by the code generators. Not sure what can be done on the generator side.

However, on NSwag, you can use the OpenApi document callback to edit the generated document to make it look the way you want.

**Comment by SebastianStehle** [comment]
I found this: https://github.com/RicoSuter/NJsonSchema/search?q=AllowReferencesWithProperties

**Comment by RicoSuter** [comment]
Yeah, we use oneof if there are more props alongside $ref (because json pointer specifies that $ref must be alone).. however many tools cannot handle this (also azure api mgmt) and you can turn it off with this cfg

**Comment by SebastianStehle** [comment]
Yes, I have forgotten to mention that it works as expected, but why is this not the default?

**Comment by RicoSuter** [comment]
> but why is this not the default?

Because it technically produces invalid JSON Schemas...
But I asked myself the same question. Changing this default would be a breaking change.
But maybe we should do that in the next major NJS/NSwag release.
